### PR TITLE
Move c-s-s to new js messaging framework

### PR DESCRIPTION
**Pull request by marcosholgado**
Task/Issue URL: https://app.asana.com/0/488551667048375/1205885707721116/f 

### Description
This PR refactors c-s-s to use the new JS Messaging framework.

### Steps to test this PR

- [ ] Privacy Test should pass, see https://github.com/duckduckgo/Android/actions/runs/6810534585
- [x] Fingerprinting is failing for a different issue so run that one locally
- [x] Test c-s-s features
- [x] When creating a new tab and going to a website if you filter by Message received you should see a new fingerprinting message

### CODE REVIEW

**Comment by marcosholgado** [comment]
Current dependencies on/for this PR:
* develop
  * **PR #3811** <a href="https://app.graphite.dev/github/pr/duckduckgo/Android/3811" target="_blank">Graphite</a>  👈

This [stack of pull requests](https://stacking.dev/?utm_source=stack-comment) is managed by [Graphite](https://app.graphite.dev/github/pr/duckduckgo/Android/3811?utm_source=stack-comment).

File: app/src/main/java/com/duckduckgo/app/browser/BrowserTabFragment.kt [hunk 1]
```diff
@@ -2100,7 +2103,16 @@ class BrowserTabFragment :
...
-            contentScopeScripts.addJsInterface(it)
+            contentScopeScripts.register(
+                it,
+                object : JsMessageCallback(this) {
+                    override fun process(method: String) {
+                        runCatching {
+                            callback.javaClass.getDeclaredMethod(method)
+                        }.getOrNull()?.invoke(callback)
+                    }
+                },
```
Comment (joshliebe) [inline app/src/main/java/com/duckduckgo/app/browser/BrowserTabFragment.kt:2114]: Is there a cleaner way of doing this? It doesn't seem like something a consumer of this method should be worrying about?

File: node_modules/@duckduckgo/content-scope-scripts/build/android/contentScope.js
[the remark this answers is not in the document]
Comment (marcosholgado) [reply node_modules/@duckduckgo/content-scope-scripts/build/android/contentScope.js:1]: Yup, once this is approved I'll merge the update css PR first and then this one to get the changes from there.

File: app/src/main/java/com/duckduckgo/app/browser/BrowserTabFragment.kt (hunk 1 above)
Comment (marcosholgado) [reply app/src/main/java/com/duckduckgo/app/browser/BrowserTabFragment.kt:2114]: This is not really used yet so I could remove it. We are using reflection to invoke the callback so we can have a single callback interface and let each handler call whichever methods they need. The idea is that this code should never change. You can see how this is used in `SubscriptionMessagingInterface`

**Review by joshliebe — APPROVED** [review]
LGTM 🚀

**Pull request merged**